File: Python/342.Power of Four.py
#判断一个数是否是4的次方
#先把不是2的次方的排除掉：num&(num-1)
class Solution(object):
    def isPowerOfFour(self, num):
        return num & (num-1)==0 and num & 0x55555555 != 0

s = Solution()
print(s.isPowerOfFour(8))

File: C++/Bit Manipulation/342.Power of Four.cpp
/*
判断一个数是不是4的次方。
*/
/*
思路：首先判断是不是2的次方。然后判断二进制位1是不是在奇数位置上。
 */
#include <iostream>
#include <vector>
using namespace std;

class Solution {
public:
	bool isPowerOfFour(int num) {
		return num > 0 && (num&(num-1))==0 && (num&0x55555555)!=0;
    }
};
int main()
{
	Solution s;
	cout<<s.isPowerOfFour(256)<<endl;
}

